src/helpers/submitPaymentTransaction.ts

```diff
@@ -1,59 +1,46 @@
-import StellarSdk, {
-  Transaction,
-  MemoType,
-  MemoValue,
-  Keypair,
-} from "stellar-sdk";
-// @ts-ignore
+import StellarSdk, { MemoType, MemoValue, Keypair } from "stellar-sdk";
 import { AuthType } from "constants/types.d";
 import { PaymentTransactionParams } from "ducks/sendTransaction";
 import { getErrorString } from "helpers/getErrorString";
 import { getNetworkConfig } from "helpers/getNetworkConfig";
-import { getTrezorSignature } from "helpers/wallet/getTrezorSignature";
 import { store } from "config/store";
+import { signTransaction } from "helpers/keyManager";
 
 export const submitPaymentTransaction = async (
   params: PaymentTransactionParams,
   authType?: AuthType,
 ) => {
-  const { settings } = store.getState();
+  const { settings, keyStore } = store.getState();
   const server = new StellarSdk.Server(
     getNetworkConfig(settings.isTestnet).url,
   );
 
-  let transaction = await buildPaymentTransaction(params);
-  transaction = await signTransaction(transaction, params, authType);
+  if (!authType) {
+    throw new Error(`Bad authentication`);
+  }
 
-  const result = await server.submitTransaction(transaction);
-  return result;
-};
+  let transaction = await buildPaymentTransaction(params);
 
-export const signTransaction = async (
-  transaction: Transaction,
-  params: PaymentTransactionParams,
-  authType?: AuthType,
-) => {
+  // Sign transaction
   try {
-    if (!authType) {
-      throw new Error(`Bad authentication`);
-    }
-
     if (authType === AuthType.PRIVATE_KEY) {
-      const keypair = Keypair.fromSecret(params.secret);
-      transaction.sign(keypair);
+      transaction = transaction.sign(Keypair.fromSecret(params.secret));
```

Comment:
Ostensibly you should be able to use `signTransaction` for plaintext keys too? The wallet SDK supports that, and that seems better than decrypting the key locally and worrying about whether it gets exposed to browser extensions, XSS attacks, etc?

Reply:
Yes, that would be nice. 👍  I'll create a ticket for that.